pkg/query-service/dao/sqlite/rbac.go

```diff
@@ -531,3 +536,53 @@ func (mds *ModelDaoSqlite) GetResetPasswordEntry(ctx context.Context,
 	}
 	return &entries[0], nil
 }
+
+// CreateUserFlags inserts user specific flags
+func (mds *ModelDaoSqlite) UpdateUserFlags(ctx context.Context, userId string, flags map[string]string) (model.UserFlag, *model.ApiError) {
+
+	if len(flags) == 0 {
+		// nothing to do as flags are empty. In this method, we only append the flags
+		// but not set them to empty
+		return flags, nil
+	}
+
+	// fetch existing flags
+	userPayload, apiError := mds.GetUser(ctx, userId)
+	if apiError != nil {
+		return nil, apiError
+	}
+
+	if userPayload.Flags != nil {
+		for k, v := range userPayload.Flags {
+			if _, ok := flags[k]; !ok {
+				// insert only missing keys as we want to retain the
+				// flags in the db that are not part of this request
+				flags[k] = v
+			}
+		}
+	}
+
+	// append existing flags with new ones
+
+	// write the updated flags
+	flagsBytes, err := json.Marshal(flags)
+	if err != nil {
+		return nil, model.InternalError(err)
+	}
+
+	if len(userPayload.Flags) == 0 {
+		q := `INSERT INTO user_flags (user_id, flags) VALUES (?, ?);`
+
+		if _, err := mds.db.ExecContext(ctx, q, userId, string(flagsBytes)); err != nil {
+			return nil, &model.ApiError{Typ: model.ErrorInternal, Err: err}
+		}
+	} else {
+		q := `UPDATE user_flags SET flags = ? WHERE user_id = ?;`
+
+		if _, err := mds.db.ExecContext(ctx, q, userId, string(flagsBytes)); err != nil {
+			return nil, &model.ApiError{Typ: model.ErrorInternal, Err: err}
+		}
+	}
```

Comment:
You could probably use `INSERT OR REPLACE` here

Reply:
avoiding it as I am not sure postgres supports it. There are alternatives in PG but in the interest of time. I would skip for now. keep it simple.